converters/toZigbee.js

```diff
@@ -288,6 +288,27 @@ const converters = {
             }
         },
     },
+    // Some devices reset brightness to 100% when turned on, even if previous brightness was different
+    // This uses the stored state of the device to restore to the previous brightness level when turning on
+    light_onoff_restorable_brightness: {
+        key: ['state', 'brightness', 'brightness_percent'],
+        convertSet: async (entity, key, value, meta) => {
+            const deviceState = meta.state || {};
+            const message = meta.message;
+            const state = message.hasOwnProperty('state') ? message.state.toLowerCase() : null;
+            const hasBrightness = message.hasOwnProperty('brightness') || message.hasOwnProperty('brightness_percent');
+
+            // Add brightness if command is 'on' and we can restore previous value
+            if (state === 'on' && !hasBrightness && deviceState.brightness > 0) {
+                message.brightness = deviceState.brightness;
```

Comment:
There is one other thing we could try before merging this PR. If this works we could get rid of the `restorable_brightness` converters and just use the existing ones.

But before implementing the required change in zigbee2mqtt, could you try the following here:

```js
await entity.command('genOnOff', 'on', {});
await entity.command('genLevelCtrl','moveToLevel', {level: deviceState.brightnes, transtime: 0});
return {state: {state: 'ON', brightness: deviceState.brightnes}};
```

Does this give the same behavior?

Reply:
With that code the dimmer flashes to 100% brightness and then quickly changes to the correct brightness when turned on. In my opinion that's a bad experience, especially considering I keep my room lights dimmed to ≈30%... with that, every time I turned them on I would get momentarily blinded 😛

Reply:
Yes I see, that's indeed not a solution, thanks for trying though!